app/src/main/java/cm/aptoide/pt/app/view/AppViewPresenter.java

```diff
@@ -213,22 +230,39 @@ private void handleOnSimilarAppsVisible() {
     view.getLifecycleEvent()
         .filter(event -> event.equals(View.LifecycleEvent.CREATE))
         .doOnNext(__ -> view.showLoading())
-        .flatMap(__ -> loadApp().flatMapCompletable(appViewViewModel -> showBannerAd())
-            .flatMapSingle(
-                appViewViewModel -> manageOrganicAds(appViewViewModel.getMinimalAd()).onErrorReturn(
-                    __1 -> null)
-                    .map(__1 -> appViewViewModel))
-            .filter(app -> app.hasDonations())// after getApk webservice is updated
-            .flatMapSingle(app -> appViewManager.getTopDonations(app.getPackageName()))
-            .observeOn(viewScheduler)
-            .doOnNext(donations -> view.showDonations(donations)))
+        .flatMap(__ -> loadApp().flatMapSingle(appViewViewModel -> showBannerAd().andThen(
+            manageOrganicAds(appViewViewModel.getMinimalAd()).onErrorReturn(__1 -> null)
+                .map(__1 -> appViewViewModel)))
+            .flatMapSingle(app -> {
+              if (app.hasDonations()) {
+                return appViewManager.getTopDonations(app.getPackageName())
+                    .observeOn(viewScheduler)
+                    .doOnSuccess(donations -> view.showDonations(donations))
+                    .map(donations -> app);
+              } else {
+                return Single.just(app);
+              }
+            }))
         .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
         .subscribe(__ -> {
         }, throwable -> {
           crashReport.log(throwable);
         });
   }
 
+  private void handleWalletPromotion() {
+    view.getLifecycleEvent()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(created -> view.isAppViewReadyToDownload())
+        .flatMapSingle(__ -> appViewManager.loadAppViewViewModel())
+        .flatMap(this::updateWalletPromotion)
```

Comment:
Don't know if you agree, but I don't agree with this name - updateWalletPromotion. Usually i only call updateXXXX if there is an initial setup and then we are constantly updating that same view, which is not what seems to be happening here.